packages/block-library/src/gallery/index.php

```diff
@@ -32,6 +32,24 @@ function block_core_gallery_data_id_backcompatibility( $parsed_block ) {
 
 add_filter( 'render_block_data', 'block_core_gallery_data_id_backcompatibility' );
 
+/**
+ * Filter to randomize the order of image blocks.
+ *
+ * @param array $parsed_block The block being rendered.
+ * @return array The block object with randomized order of image blocks.
+ */
+function block_core_gallery_random_order( $parsed_block ) {
+	if ( 'core/gallery' === $parsed_block['blockName'] && isset( $parsed_block['attrs']['randomOrder'] ) && $parsed_block['attrs']['randomOrder'] ) {
+		$inner_blocks = $parsed_block['innerBlocks'];
+		shuffle( $inner_blocks );
+		$parsed_block['innerBlocks'] = $inner_blocks;
+	}
```

Comment:
```suggestion
	if ( 'core/gallery' === $parsed_block['blockName'] && ! empty( $parsed_block['attrs']['randomOrder'] ) ) {
		shuffle( $parsed_block['innerBlocks'] );
	}

	return $parsed_block;
```

We could make this a little less verbose given we're not checking strictly for `true` and are already mutating `$parsed_block`.

What do you think?

Reply:
I definitely think it's simpler and easier to understand 👍